### DISCUSSION TITLE
Is there way to take backup for multiple endpoints

### DISCUSSION THREAD

**Discussion by kavita1205** [etcd-io/etcd#14643] [Q&A] [answered]
Hi Team,

We are using below command to take etcd backup but this command works only in case of single node. Do we have any way in case we want to take backup for multiple nodes, meaning how we can provide multiple endpoints.

ETCDCTL_API=3 etcdctl --endpoints=https://$IP:2379 --cacert=/etc/kubernetes/pki/etcd/ca.crt --cert=/etc/kubernetes/pki/etcd/client.crt --key=/etc/kubernetes/pki/etcd/client.key snapshot save /var/etcd-backup/etcd-snapshot-$(date +%Y-%m-%dT%H:%M).db

**Comment by tjungblu** [comment, marked as answer]
not really, but you can just write a bash script wrapper around this easily with a bit of json manipulation:

```
etcdctl member list -wjson | jq -r '.members[].clientURLs | .[]' | while read member ; do
   etcdctl --endpoints=${member} snapshot save  .... 
done
```
